06_timers/time_module/get_time_module.c

```diff
@@ -0,0 +1,110 @@
+// SPDX-License-Identifier: GPL-3.0
+/**/
+
+#define pr_fmt(fmt) "%s: " fmt,  KBUILD_MODNAME
+#include <linux/kobject.h>
+#include <linux/string.h>
+#include <linux/sysfs.h>
+#include <linux/module.h>
+#include <linux/init.h>
+#include <linux/ctype.h>
+
+#define LEN_STR 160
+
+static char buf_str[LEN_STR + 1];
+static struct timespec64 r_last_time;
+static struct timespec64 a_last_time;
+static bool first_read = 1;
+
+static ssize_t relation_delay_show(struct kobject *kobj,
+	struct kobj_attribute *attr, char *buf)
+{
+	int ret = 0;
+	struct timespec64 r_new_time;
+
+	if (first_read) {
+		ret = scnprintf(buf, PAGE_SIZE,
+		"RELATION DELAY FROM LAST READING \t 0s\n");
+		ktime_get_real_ts64(&a_last_time);
+		ktime_get_ts64(&r_new_time);
+		first_read = 0;
+		return ret;
+	}
+	ktime_get_real_ts64(&a_last_time);
+	ktime_get_ts64(&r_new_time);
+	ret = scnprintf(buf, PAGE_SIZE,
+		"RELATION DELAY FROM LAST READING \t %lld.%09lds\n",
+		(s64)r_new_time.tv_sec-(s64)r_last_time.tv_sec,
+		r_new_time.tv_nsec-r_last_time.tv_nsec);
```

Comment:
This conversation is correct only for values where  r_new_time.tv_nsec >= r_last_time.tv_nsec.

Reply:
I`ve fixed